lib/ecto/adapters/sql.ex

```diff
@@ -348,63 +345,43 @@ defmodule Ecto.Adapters.SQL do
     end
   end
 
-  defp begin_sandbox(repo, ref, queue_time, timeout, opts) do
-    log? = Keyword.get(opts, :log, true)
-    case begin(ref, :raw, 1, queue_time, log?, timeout, opts) do
+  defp begin_sandbox(repo, ref, mod, queue_time, timeout, opts) do
+    case begin(repo, mod, :raw, 1, queue_time, opts) do
       {{:ok, _}, entry} ->
-        log_and_savepoint = fn() ->
-          log(repo, entry)
-          savepoint_sandbox(ref, log?, timeout, opts)
-        end
-        Pool.fuse(ref, timeout, log_and_savepoint)
-      {{:error, _err}, _entry} = error ->
+        savepoint_sandbox(repo, ref, mod, entry, timeout, opts)
+      {{:error, _}, _entry} = error ->
         Pool.break(ref, timeout)
         error
     end
   end
 
-  defp savepoint_sandbox(ref, log?, timeout, opts) do
-    case begin(ref, :raw, :sandbox, nil, log?, timeout, opts) do
-      {{:ok, _}, entry} ->
+  defp savepoint_sandbox(repo, ref, mod, entry, timeout, opts) do
+    case begin(repo, mod, :raw, :sandbox, nil, opts) do
+      # We ignore the entry, keeping only the begin transaction one
```

Comment:
:worried:, why skip the first and not the second? The second makes it clear is a sandbox.